mods/sethome/init.lua

```diff
@@ -24,42 +25,69 @@ loadhomes()
 
 minetest.register_privilege("home", "Can use /sethome and /home")
 
-local changed = false
+function sethome.go_home(name)
+	local player = minetest.get_player_by_name(name)
+	if not player then
+		-- just a check to prevent the server crashing
+		return false
+	end
+	if homepos[name] then
+		player:setpos(homepos[name])
+		return true
+	else
+		return false
+	end
+end
 
 minetest.register_chatcommand("home", {
-    description = "Teleport you to your home point",
-    privs = {home=true},
-    func = function(name)
-        local player = minetest.get_player_by_name(name)
-        if player == nil then
-            -- just a check to prevent the server crashing
-            return false
-        end
-        if homepos[player:get_player_name()] then
-            player:setpos(homepos[player:get_player_name()])
-            minetest.chat_send_player(name, "Teleported to home!")
-        else
-            minetest.chat_send_player(name, "Set a home using /sethome")
-        end
-    end,
+	description = "Teleport you to your home point",
+	privs = {home=true},
+	func = function(name)
+		local res = sethome.go_home(name)
+		if res then
+			return true, "Teleported to home!"
+		else
+			return false, "Set a home using /sethome"
+		end
+	end,
 })
 
+function sethome.set_home(name, pos)
+	local player = minetest.get_player_by_name(name)
+	if not player then
```

Comment:
Should probably not forbid setting home of offline players.

Suggestion:

```lua
if type(pos) ~= "table" then -- Checking for `nil` is redundant since type(nil) == "nil"
  local player = minetest.get_player_by_name(name)
  if player then
    pos = player:getpos()
  else
    return false, "no position given and player is offline"
  end
end
```

Alternatively, just outright require the `pos` parameter:
```lua
assert(type(pos)=="table", "bad `pos` argument to `set_home`: table expected got "..type(pos))
```